src/key_packages/mod.rs

```diff
@@ -72,11 +71,11 @@ impl KeyPackage {
             protocol_version: ProtocolVersion::default(),
             cipher_suite: ciphersuite_name,
             hpke_init_key,
-            credential,
+            credential: credential_bundle.credential().clone(),
             extensions,
             signature: Signature::new_empty(),
         };
-        key_package.sign(&ciphersuite, signature_key);
+        key_package.sign(&ciphersuite, credential_bundle.signature_private_key());
```

Comment:
So we pass in a `CredentialBundle` but then only consume the credential? This doesn't look like what we should be doing.